sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -469,19 +442,14 @@ static void send_device_info(void)
 static void send_serial_number(void)
 {
   // Get the property PATCH topic to send a reported property update.
-  az_result rc = az_iot_pnp_client_property_patch_get_publish_topic(
-      &pnp_client,
-      pnp_mqtt_get_request_id(),
-      publish_message.topic,
-      publish_message.topic_length,
-      NULL);
-
-  if (az_result_failed(rc))
-  {
-    IOT_SAMPLE_LOG_ERROR(
-        "Failed to get the property PATCH topic: az_result return code 0x%08x.", rc);
-    exit(rc);
-  }
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(
+      az_iot_pnp_client_property_patch_get_publish_topic(
```

Comment:
On some of these longer function calls, would it be easier to read if the call was made first (as prior), and then the result passed in to the macro?

```
az_result rc = az_iot_pnp_client_property_patch_get_publish_topic(
      &pnp_client,
      pnp_mqtt_get_request_id(),
      publish_message.topic,
      publish_message.topic_length,
      NULL);
IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Failed to get the property PATCH topic");
```

Reply:
yes great call. Will update for cases that maybe don't have chained/dependent successive calls.